packages/treasury/src/vaultManager.js

```diff
@@ -84,42 +83,52 @@ export function makeVaultManager(
   // when a priceQuote is received, we'll only reschedule if the high-water
   // level when the request was made matches the current high-water level.
   async function reschedulePriceCheck() {
-    const highestRatioWhenScheduled = sortedVaultKits.highestRatio();
-    if (!highestRatioWhenScheduled) {
+    const highestDebtRatio = sortedVaultKits.highestRatio();
+    if (!highestDebtRatio) {
       // if there aren't any open vaults, we don't need an outstanding RFQ.
       return;
     }
 
-    highestDebtRatio = highestRatioWhenScheduled;
     const liquidationMargin = shared.getLiquidationMargin();
 
-    // We ask to be alerted when the price level falls enough that the vault
+    // ask to be alerted when the price level falls enough that the vault
     // with the highest debt to collateral ratio will no longer be valued at the
     // liquidationMargin above its debt.
     const triggerPoint = multiplyBy(
-      highestRatioWhenScheduled.numerator,
+      highestDebtRatio.numerator,
       liquidationMargin,
     );
-    // Notice that this is scheduleing a callback for later (possibly much
-    // later). Callers shouldn't be expecting a response from this function.
-    const quote = await E(priceAuthority).quoteWhenLT(
-      highestRatioWhenScheduled.denominator,
+
+    // if there's an outstanding quote, reset the level. If there's no current
+    // quote (because this is the first loan, or because a quote just resolved)
+    // then make a new request to the priceAuthority, and when it resolves,
+    // liquidate anything that's above the price level.
+    if (outstandingQuote) {
+      E(outstandingQuote).updateLevel(
```

Comment:
Whenever a new quote is received (line 123), we will call `.forEachRatioGTE(quoteRatioPlusMargin)`. That will liquidate any vaults that are above the then-current margin, regardless of where the margin was when the request was scheduled. This might be zero or more than expected when it was scheduled. After we receive the new quote, we'll schedule another one, regardless of whether we liquidated anything or if liquidity ratios have changed.

I think we may receive quotes that don't correspond to the current highest ratio, but we'll handle that correctly. We'll always have a quote outstanding if there are any active vaults.

Reply:
Great answer!